src/ai/backend/manager/api/gql/agent/types.py

```diff
@@ -0,0 +1,53 @@
+from datetime import datetime
+from typing import Optional
+
+import strawberry
+from strawberry.relay import Node, NodeID
+from strawberry.scalars import JSON
+
+from ai.backend.manager.data.agent.types import AgentStatus
+from ai.backend.manager.models.rbac.permission_defs import AgentPermission
+
+
+@strawberry.type(description="Resource-related information for an agent")
+class AgentResourceInfo:
+    available_slots: JSON
+    occupied_slots: JSON
+    compute_plugins: JSON
+
+
+@strawberry.type(description="Status and lifecycle information for an agent")
+class AgentStatusInfo:
+    status: AgentStatus
+    status_changed: datetime
+    first_contact: datetime
+    lost_at: Optional[datetime]
+    schedulable: bool
+
+
+@strawberry.type(description="System and configuration information for an agent")
+class AgentSystemInfo:
+    architecture: str
+    version: str
+    auto_terminate_abusing_kernel: bool = strawberry.field(
+        description="Not used anymore, present for schema consistency."
+    )
```

Comment:
All field descriptions must be included.
Please try to write multi-line descriptions if possible.
Please include the applied version in the type description as well. (Added in 25.18.0)